controller/src/main/java/io/pravega/controller/store/stream/PersistentStreamBase.java

```diff
@@ -1442,6 +1465,10 @@ private TxnStatus handleDataNotFoundException(Throwable ex) {
         }).collect(Collectors.toList()));
     }
 
+    private int getShardNumber(long segmentId) {
```

Comment:
@fpj 
There were two causes for `DataNotFoundException` while accessing `sealedSegmentSizesMapShard` 
1. getShardNumber was using `segmentNumber` instead of `epoch`
2. If shard was not present, because no segment mapped to the shard was sealed, we now return empty map. 

`1` was causing the existing unit test to fail (this was observed in travis build)
`2` was found through new targeted unit test.